core/io/multiplayer_peer_custom.cpp

```diff
@@ -0,0 +1,184 @@
+}
+
+int MultiplayerPeerCustom::get_max_packet_size() const {
+	return max_packet_size;
+}
+
+//
+// NetworkedMultiplayerPeer.
+//
+
+void MultiplayerPeerCustom::set_transfer_mode(NetworkedMultiplayerPeer::TransferMode p_mode) {
+	transfer_mode = p_mode;
+}
+
+NetworkedMultiplayerPeer::TransferMode MultiplayerPeerCustom::get_transfer_mode() const {
+	return transfer_mode;
+}
+
+void MultiplayerPeerCustom::set_target_peer(int p_target_peer) {
+	target_id = p_target_peer;
+}
+
+int MultiplayerPeerCustom::get_packet_peer() const {
+	ERR_FAIL_COND_V(connection_status != ConnectionStatus::CONNECTION_CONNECTED, 1);
+	ERR_FAIL_COND_V(incoming_packets.size() == 0, 1);
+
+	return incoming_packets.front()->get().from;
+}
+
+bool MultiplayerPeerCustom::is_server() const {
+	return self_id == 1;
+}
+
+void MultiplayerPeerCustom::poll() {
+}
+
+int MultiplayerPeerCustom::get_unique_id() const {
+	return self_id;
+}
+
+void MultiplayerPeerCustom::set_refuse_new_connections(bool p_refuse_new_connections) {
+	refusing_new_connections = p_refuse_new_connections;
+}
+
+bool MultiplayerPeerCustom::is_refusing_new_connections() const {
+	return refusing_new_connections;
+}
+
+NetworkedMultiplayerPeer::ConnectionStatus MultiplayerPeerCustom::get_connection_status() const {
+	return connection_status;
+}
+
+//
+// Custom methods.
+//
+
+void MultiplayerPeerCustom::initialize(int p_self_id) {
+	if (connection_status != ConnectionStatus::CONNECTION_CONNECTING) {
+		return;
+	}
+
+	self_id = p_self_id;
+	if (self_id == 1) {
+		set_connection_status(ConnectionStatus::CONNECTION_CONNECTED);
+	}
+}
+
+void MultiplayerPeerCustom::set_max_packet_size(int p_max_packet_size) {
+	max_packet_size = p_max_packet_size;
+}
+
+void MultiplayerPeerCustom::set_connection_status(NetworkedMultiplayerPeer::ConnectionStatus p_connection_status) {
+	// Can only go to connecting, if we are disconnected.
+	if (p_connection_status == ConnectionStatus::CONNECTION_CONNECTING && connection_status == ConnectionStatus::CONNECTION_DISCONNECTED) {
+		connection_status = p_connection_status;
+	}
+	// Can only go to connected, if we are connecting.
+	else if (p_connection_status == ConnectionStatus::CONNECTION_CONNECTED && connection_status == ConnectionStatus::CONNECTION_CONNECTING) {
+		connection_status = p_connection_status;
+		emit_signal("connection_succeeded");
+	}
+	// Can only go to disconnected, if we aren't already disconnected.
+	else if (p_connection_status == ConnectionStatus::CONNECTION_DISCONNECTED && connection_status != ConnectionStatus::CONNECTION_DISCONNECTED) {
+		ConnectionStatus old_connection_status = connection_status;
+		connection_status = p_connection_status;
+
+		if (self_id != 1) {
+			if (old_connection_status == ConnectionStatus::CONNECTION_CONNECTING) {
+				emit_signal("connection_failed");
+			} else {
+				emit_signal("server_disconnected");
+			}
+		}
+	}
+}
+
+void MultiplayerPeerCustom::deliver_rpc(const PoolByteArray &p_data, int p_from_peer_id) {
```

Comment:
Similarly as above, this might deliver things other than RPCs, so something like `queue_packet` or `insert_packet` might be more appropriate.